v2/elasticsearch-common/src/main/java/com/google/cloud/teleport/v2/elasticsearch/utils/ElasticsearchIO.java

```diff
@@ -1528,10 +1528,11 @@ static int getBackendVersion(ConnectionConfiguration connectionConfiguration) {
           (backendVersion == 2
               || backendVersion == 5
               || backendVersion == 6
-              || backendVersion == 7),
+              || backendVersion == 7
+              || backendVersion == 8),
```

Comment:
Hey @suyograo, thanks for taking a look. This line adds 8 as an allowed version to the [checkArgument](https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html) above on line 1527 which when true returns the backendVersion instead of throwing the error below it. The only issue is this checkArgument that only allows specific versions.

Reply:
Indeed both check and error message has been changed